diff --git a/geshi-extra/igor.php b/geshi-extra/igor.php
new file mode 100644
index 0000000..d28df27
--- /dev/null
+++ b/geshi-extra/igor.php
@@ -0,0 +1,385 @@
+<?php
+/*************************************************************************************
+ * HardDrive:Users:aclight:Documents:scripts:igor.php
+ * --------
+ * Author: Adam Light (aclight@gmail.org)
+ * Copyright: (c) 2004 Nigel McNie (http://qbnz.com/highlighter/)
+ * Release Version: 1.0.7.18
+ * Date Started: 2007/02/25
+ *
+ * IGOR language file for GeSHi.
+ *
+ * CURRENT LANGUAGE VERSION
+ * Created on: Mon, Jun 18, 2012
+ * Igor Version: 6.23B04
+ *
+ *************************************************************************************
+ *
+ *     This file is part of GeSHi.
+ *
+ *   GeSHi is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *   GeSHi is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with GeSHi; if not, write to the Free Software
+ *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ *
+ ************************************************************************************/
+
+$language_data = array (
+	'LANG_NAME' => 'IGOR',
+	'COMMENT_SINGLE' => array(1 => '//'),
+	'COMMENT_MULTI' => array(),
+	'CASE_KEYWORDS' => GESHI_CAPS_NO_CHANGE,
+	'QUOTEMARKS' => array('"'),
+	'ESCAPE_CHAR' => '\\',
+	'KEYWORDS' => array(
+		1 => array(
+			'while', 'try', 'switch', 'strswitch', 
+			'return', 'if', 'for', 'endtry', 
+			'endswitch', 'endif', 'endfor', 'elseif', 
+			'else', 'do', 'continue', 'catch', 
+			'case', 'break'
+			),
+		2 => array(
+			'window', 'WAVE', 'variable', 'ThreadSafe', 
+			'SVAR', 'SubMenu', 'Structure', 'STRUCT', 
+			'string', 'strconstant', 'static', 'Prompt', 
+			'Proc', 'Picture', 'NVAR', 'MultiThread', 
+			'Menu', 'macro', 'graph', 'function', 
+			'FuncFit', 'EndStructure', 'EndMacro', 'end', 
+			'DoPrompt', 'constant'
+			),
+		3 => array(
+			'ZernikeR', 'zcsr', 'z', 'y', 
+			'XWaveRefFromTrace', 'XWaveName', 'xcsr', 'x2pnt', 
+			'x', 'WNoise', 'WinType', 'WinRecreation', 
+			'WinName', 'WinList', 'WhichListItem', 'WaveUnits', 
+			'WaveType', 'WaveRefsEqual', 'WaveRefIndexed', 'WaveName', 
+			'WaveMin', 'WaveMax', 'WaveList', 'WaveInfo', 
+			'WaveExists', 'WaveDims', 'WaveCRC', 'vcsr', 
+			'Variance', 'VariableList', 'URLEncode', 'URLDecode', 
+			'UpperStr', 'UnPadString', 'UniqueName', 'trunc', 
+			'TraceNameToWaveRef', 'TraceNameList', 'TraceInfo', 'TraceFromPixel', 
+			'time', 'ticks', 'ThreadReturnValue', 'ThreadProcessorCount', 
+			'ThreadGroupWait', 'ThreadGroupRelease', 'ThreadGroupGetDFR', 'ThreadGroupGetDF', 
+			'ThreadGroupCreate', 'TextFile', 'tanh', 'tan', 
+			'TagWaveRef', 'TagVal', 'TableInfo', 't', 
+			'SVAR_Exists', 'sum', 'StudentT', 'StudentA', 
+			'StrVarOrDefault', 'strsearch', 'strlen', 'stringmatch', 
+			'StringList', 'StringFromList', 'stringCRC', 'StringByKey', 
+			'str2num', 'StopMSTimer', 'StatsWeibullPDF', 'StatsWeibullCDF', 
+			'StatsWaldPDF', 'StatsWaldCDF', 'StatsVonMisesPDF', 'StatsVonMisesNoise', 
+			'StatsVonMisesCDF', 'StatsUSquaredCDF', 'StatsTrimmedMean', 'StatsTriangularPDF', 
+			'StatsTriangularCDF', 'StatsTopDownCDF', 'StatsStudentPDF', 'StatsStudentCDF', 
+			'StatsSpearmanRhoCDF', 'StatsRunsCDF', 'StatsRectangularPDF', 'StatsRectangularCDF', 
+			'StatsRayleighPDF', 'StatsRayleighCDF', 'StatsQpCDF', 'StatsQCDF', 
+			'StatsPowerPDF', 'StatsPowerNoise', 'StatsPowerCDF', 'StatsPoissonPDF', 
+			'StatsPoissonCDF', 'StatsPermute', 'StatsParetoPDF', 'StatsParetoCDF', 
+			'StatsNormalPDF', 'StatsNormalCDF', 'StatsNCTPDF', 'StatsNCTCDF', 
+			'StatsNCFPDF', 'StatsNCFCDF', 'StatsNCChiPDF', 'StatsNCChiCDF', 
+			'StatsNBinomialPDF', 'StatsNBinomialCDF', 'StatsMooreCDF', 'StatsMedian', 
+			'StatsMaxwellPDF', 'StatsMaxwellCDF', 'StatsLogNormalPDF', 'StatsLogNormalCDF', 
+			'StatsLogisticPDF', 'StatsLogisticCDF', 'StatsKuiperCDF', 'StatsInvWeibullCDF', 
+			'StatsInvVonMisesCDF', 'StatsInvUsquaredCDF', 'StatsInvTriangularCDF', 'StatsInvTopDownCDF', 
+			'StatsInvStudentCDF', 'StatsInvSpearmanCDF', 'StatsInvRectangularCDF', 'StatsInvRayleighCDF', 
+			'StatsInvQpCDF', 'StatsInvQCDF', 'StatsInvPowerCDF', 'StatsInvPoissonCDF', 
+			'StatsInvParetoCDF', 'StatsInvNormalCDF', 'StatsInvNCFCDF', 'StatsInvNCChiCDF', 
+			'StatsInvNBinomialCDF', 'StatsInvMooreCDF', 'StatsInvMaxwellCDF', 'StatsInvLogNormalCDF', 
+			'StatsInvLogisticCDF', 'StatsInvKuiperCDF', 'StatsInvGeometricCDF', 'StatsInvGammaCDF', 
+			'StatsInvFriedmanCDF', 'StatsInvFCDF', 'StatsInvExpCDF', 'StatsInvEValueCDF', 
+			'StatsInvDExpCDF', 'StatsInvCMSSDCDF', 'StatsInvChiCDF', 'StatsInvCauchyCDF', 
+			'StatsInvBinomialCDF', 'StatsInvBetaCDF', 'StatsHyperGPDF', 'StatsHyperGCDF', 
+			'StatsGeometricPDF', 'StatsGeometricCDF', 'StatsGammaPDF', 'StatsGammaCDF', 
+			'StatsFriedmanCDF', 'StatsFPDF', 'StatsFCDF', 'StatsExpPDF', 
+			'StatsExpCDF', 'StatsEValuePDF', 'StatsEValueCDF', 'StatsErrorPDF', 
+			'StatsErlangPDF', 'StatsErlangCDF', 'StatsDExpPDF', 'StatsDExpCDF', 
+			'StatsCorrelation', 'StatsCMSSDCDF', 'StatsChiPDF', 'StatsChiCDF', 
+			'StatsCauchyPDF', 'StatsCauchyCDF', 'StatsBinomialPDF', 'StatsBinomialCDF', 
+			'StatsBetaPDF', 'StatsBetaCDF', 'StartMSTimer', 'sqrt', 
+			'SphericalHarmonics', 'SphericalBessYD', 'SphericalBessY', 'SphericalBessJD', 
+			'SphericalBessJ', 'SpecialDirPath', 'SpecialCharacterList', 'SpecialCharacterInfo', 
+			'SortList', 'sinh', 'sinc', 'sin', 
+			'sign', 'SelectString', 'SelectNumber', 'Secs2Time', 
+			'Secs2Date', 'sec', 'ScreenResolution', 'sawtooth', 
+			's', 'round', 'rightx', 'ReplaceStringByKey', 
+			'ReplaceString', 'ReplaceNumberByKey', 'RemoveListItem', 'RemoveFromList', 
+			'RemoveEnding', 'RemoveByKey', 'real', 'r2polar', 
+			'r', 'qcsr', 'q', 'ProcedureText', 
+			'PossiblyQuoteName', 'PolygonArea', 'poly2D', 'poly', 
+			'poissonNoise', 'pnt2x', 'PixelFromAxisVal', 'PICTList', 
+			'PICTInfo', 'Pi', 'pcsr', 'PathList', 
+			'ParseFilePath', 'ParamIsDefault', 'PadString', 'p2rect', 
+			'p', 'OperationList', 'NVAR_Exists', 'NumVarOrDefault', 
+			'numtype', 'numpnts', 'NumberByKey', 'num2str', 
+			'num2istr', 'num2char', 'note', 'norm', 
+			'NewFreeWave', 'NewFreeDataFolder', 'NaN', 'NameOfWave', 
+			'ModDate', 'mod', 'min', 'mean', 
+			'max', 'MatrixTrace', 'MatrixRank', 'MatrixDot', 
+			'MatrixDet', 'MarcumQ', 'MandelbrotPoint', 'magsqr', 
+			'MacroList', 'LowerStr', 'lorentzianNoise', 'logNormalNoise', 
+			'log', 'ln', 'ListMatch', 'limit', 
+			'LegendreA', 'leftx', 'LayoutInfo', 'LaguerreGauss', 
+			'LaguerreA', 'Laguerre', 'JulianToDate', 'jlim', 
+			'j', 'ItemsInList', 'inverseERFC', 'inverseERF', 
+			'Interp3D', 'Interp2D', 'interp', 'Integrate1D', 
+			'Inf', 'IndexedFile', 'IndexedDir', 'IndependentModuleList', 
+			'ImageNameToWaveRef', 'ImageNameList', 'ImageInfo', 'imag', 
+			'ilim', 'IgorVersion', 'IgorInfo', 'i', 
+			'HyperGPFQ', 'HyperGNoise', 'HyperG2F1', 'HyperG1F1', 
+			'HyperG0F1', 'hermiteGauss', 'hermite', 'hcsr', 
+			'Hash', 'GuideNameList', 'GuideInfo', 'GrepString', 
+			'GrepList', 'gnoise', 'GetWavesDataFolderDFR', 'GetWavesDataFolder', 
+			'GetUserData', 'GetScrapText', 'GetRTStackInfo', 'GetRTLocInfo', 
+			'GetRTLocation', 'GetRTError', 'GetRTErrMessage', 'GetKeyState', 
+			'GetIndexedObjNameDFR', 'GetIndexedObjName', 'GetIndependentModuleName', 'GetFormula', 
+			'GetErrMessage', 'GetDimLabel', 'GetDefaultFontStyle', 'GetDefaultFontSize', 
+			'GetDefaultFont', 'GetDataFolderDFR', 'GetDataFolder', 'gcd', 
+			'Gauss2D', 'Gauss1D', 'Gauss', 'gammq', 
+			'gammp', 'gammln', 'gammaNoise', 'gammaInc', 
+			'gamma', 'FunctionPath', 'FunctionList', 'FunctionInfo', 
+			'FuncRefInfo', 'FresnelSin', 'FresnelCos', 'FontSizeStringWidth', 
+			'FontSizeHeight', 'FontList', 'floor', 'FindListItem', 
+			'FindDimLabel', 'FetchURL', 'faverageXY', 'faverage', 
+			'fakedata', 'factorial', 'expNoise', 'expInt', 
+			'exp', 'exists', 'erfcw', 'erfc', 
+			'erf', 'equalWaves', 'enoise', 'ei', 
+			'e', 'DimSize', 'DimOffset', 'DimDelta', 
+			'digamma', 'deltax', 'defined', 'DDETerminate', 
+			'DDEStatus', 'DDERequestWave', 'DDERequestString', 'DDEPokeWave', 
+			'DDEPokeString', 'DDEInitiate', 'DDEExecute', 'Dawson', 
+			'DateToJulian', 'datetime', 'date2secs', 'date', 
+			'DataFolderRefStatus', 'DataFolderRefsEqual', 'DataFolderExists', 'DataFolderDir', 
+			'CTabList', 'CsrXWaveRef', 'CsrXWave', 'CsrWaveRef', 
+			'CsrWave', 'CsrInfo', 'csc', 'CreationDate', 
+			'cpowi', 'CountObjectsDFR', 'CountObjects', 'coth', 
+			'cot', 'cosh', 'cos', 'ControlNameList', 
+			'ContourZ', 'ContourNameToWaveRef', 'ContourNameList', 'ContourInfo', 
+			'conj', 'cmpstr', 'cmplx', 'CleanupName', 
+			'ChildWindowList', 'CheckName', 'chebyshevU', 'chebyshev', 
+			'char2num', 'cequal', 'ceil', 'CaptureHistoryStart', 
+			'CaptureHistory', 'cabs', 'binomialNoise', 'binomialln', 
+			'binomial', 'BinarySearchInterp', 'BinarySearch', 'betai', 
+			'beta', 'bessy', 'bessk', 'bessj', 
+			'bessi', 'Bessely', 'Besselk', 'Besselj', 
+			'Besseli', 'AxisValFromPixel', 'AxisList', 'AxisInfo', 
+			'atanh', 'atan2', 'atan', 'asinh', 
+			'asin', 'areaXY', 'area', 'AnnotationList', 
+			'AnnotationInfo', 'alog', 'AiryBD', 'AiryB', 
+			'AiryAD', 'AiryA', 'AddListItem', 'acosh', 
+			'acos', 'abs'
+			),
+		4 => array(
+			'WindowFunction', 'WignerTransform', 'wfprintf', 'WaveTransform', 
+			'WaveStats', 'WaveMeanStdv', 'Variable', 'ValDisplay', 
+			'Unwrap', 'Triangulate3d', 'ToolsGrid', 'ToCommandLine', 
+			'TitleBox', 'TileWindows', 'Tile', 'ThreadStart', 
+			'ThreadGroupPutDF', 'TextBox', 'Tag', 'TabControl', 
+			'StructPut', 'StructGet', 'String', 'StatsWRCorrelationTest', 
+			'StatsWilcoxonRankTest', 'StatsWheelerWatsonTest', 'StatsWatsonWilliamsTest', 'StatsWatsonUSquaredTest', 
+			'StatsVariancesTest', 'StatsTukeyTest', 'StatsTTest', 'StatsSRTest', 
+			'StatsSignTest', 'StatsScheffeTest', 'StatsSample', 'StatsResample', 
+			'StatsRankCorrelationTest', 'StatsQuantiles', 'StatsNPNominalSRTest', 'StatsNPMCTest', 
+			'StatsMultiCorrelationTest', 'StatsLinearRegression', 'StatsLinearCorrelationTest', 'StatsKWTest', 
+			'StatsKSTest', 'StatsKendallTauTest', 'StatsJBTest', 'StatsHodgesAjneTest', 
+			'StatsFTest', 'StatsFriedmanTest', 'StatsDunnettTest', 'StatsDIPTest', 
+			'StatsContingencyTable', 'StatsCochranTest', 'StatsCircularTwoSampleTest', 'StatsCircularMoments', 
+			'StatsCircularMeans', 'StatsCircularCorrelationTest', 'StatsChiTest', 'StatsANOVA2Test', 
+			'StatsANOVA2RMTest', 'StatsANOVA2NRTest', 'StatsANOVA1Test', 'StatsAngularDistanceTest', 
+			'StackWindows', 'Stack', 'sscanf', 'sprintf', 
+			'SplitString', 'SphericalTriangulate', 'SphericalInterpolate', 'SoundInStopChart', 
+			'SoundInStatus', 'SoundInStartChart', 'SoundInSet', 'SoundInRecord', 
+			'Sort', 'SmoothCustom', 'Smooth', 'Slider', 
+			'Sleep', 'Silent', 'ShowTools', 'ShowInfo', 
+			'ShowIgorMenus', 'SetWindow', 'SetWaveLock', 'SetVariable', 
+			'SetScale', 'SetRandomSeed', 'SetProcessSleep', 'SetMarquee', 
+			'SetIgorOption', 'SetIgorMenuMode', 'SetIgorHook', 'SetFormula', 
+			'SetFileFolderInfo', 'SetDrawLayer', 'SetDrawEnv', 'SetDimLabel', 
+			'SetDataFolder', 'SetDashPattern', 'SetBackground', 'SetAxis', 
+			'SetActiveSubwindow', 'SaveTableCopy', 'SavePICT', 'SavePackagePreferences', 
+			'SaveNotebook', 'SaveGraphCopy', 'SaveExperiment', 'SaveData', 
+			'Save', 'Rotate', 'Reverse', 'ResumeUpdate', 
+			'Resample', 'ReplaceWave', 'ReplaceText', 'ReorderTraces', 
+			'ReorderImages', 'RenameWindow', 'RenamePICT', 'RenamePath', 
+			'RenameDataFolder', 'Rename', 'RemovePath', 'RemoveLayoutObjects', 
+			'RemoveImage', 'RemoveFromTable', 'RemoveFromLayout', 'RemoveFromGraph', 
+			'RemoveContour', 'Remove', 'Redimension', 'RatioFromNumber', 
+			'Quit', 'pwd', 'PutScrapText', 'PulseStats', 
+			'Project', 'PrintTable', 'PrintSettings', 'PrintNotebook', 
+			'PrintLayout', 'PrintGraphs', 'printf', 'Print', 
+			'PrimeFactors', 'Preferences', 'PopupMenu', 'PopupContextualMenu', 
+			'PlaySound', 'PlaySnd', 'PlayMovieAction', 'PlayMovie', 
+			'PCA', 'PauseUpdate', 'PauseForUser', 'PathInfo', 
+			'ParseOperationTemplate', 'Optimize', 'OpenNotebook', 'Open', 
+			'NotebookAction', 'Notebook', 'Note', 'NewWaterfall', 
+			'NewPath', 'NewPanel', 'NewNotebook', 'NewMovie', 
+			'NewLayout', 'NewImage', 'NewFreeAxis', 'NewFIFOChan', 
+			'NewFIFO', 'NewDataFolder', 'NeuralNetworkTrain', 'NeuralNetworkRun', 
+			'MoveWindow', 'MoveWave', 'MoveVariable', 'MoveSubwindow', 
+			'MoveString', 'MoveFolder', 'MoveFile', 'MoveDataFolder', 
+			'ModifyWaterfall', 'ModifyTable', 'ModifyPanel', 'ModifyLayout', 
+			'ModifyImage', 'ModifyGraph', 'ModifyFreeAxis', 'ModifyControlList', 
+			'ModifyControl', 'ModifyContour', 'Modify', 'MeasureStyledText', 
+			'MatrixTranspose', 'MatrixSVD', 'MatrixSVBkSub', 'MatrixSolve', 
+			'MatrixSchur', 'MatrixOP', 'MatrixMultiply', 'MatrixLUD', 
+			'MatrixLUBkSub', 'MatrixLLS', 'MatrixLinearSolveTD', 'MatrixLinearSolve', 
+			'MatrixInverse', 'MatrixGaussJ', 'MatrixFilter', 'MatrixEigenV', 
+			'MatrixCorr', 'MatrixConvolve', 'MarkPerfTestTime', 'MakeIndex', 
+			'Make', 'LombPeriodogram', 'Loess', 'LoadWave', 
+			'LoadPICT', 'LoadPackagePreferences', 'LoadData', 'ListBox', 
+			'LinearFeedbackShiftRegister', 'Legend', 'Layout', 'Label', 
+			'KMeans', 'KillWindow', 'KillWaves', 'KillVariables', 
+			'KillStrings', 'KillPICTs', 'KillPath', 'KillFreeAxis', 
+			'KillFIFO', 'KillDataFolder', 'KillControl', 'KillBackground', 
+			'Interpolate3D', 'Interp3DPath', 'IntegrateODE', 'Integrate', 
+			'InsertPoints', 'IndexSort', 'ImageWindow', 'ImageUnwrapPhase', 
+			'ImageTransform', 'ImageThreshold', 'ImageStats', 'ImageSnake', 
+			'ImageSeedFill', 'ImageSave', 'ImageRotate', 'ImageRestore', 
+			'ImageRemoveBackground', 'ImageRegistration', 'ImageMorphology', 'ImageLoad', 
+			'ImageLineProfile', 'ImageInterpolate', 'ImageHistogram', 'ImageHistModification', 
+			'ImageGenerateROIMask', 'ImageFromXYZ', 'ImageFocus', 'ImageFilter', 
+			'ImageFileInfo', 'ImageEdgeDetection', 'ImageBoundaryToMask', 'ImageBlend', 
+			'ImageAnalyzeParticles', 'IFFT', 'Histogram', 'HilbertTransform', 
+			'HideTools', 'HideProcedures', 'HideInfo', 'HideIgorMenus', 
+			'Hanning', 'GroupBox', 'Grep', 'GraphWaveEdit', 
+			'GraphWaveDraw', 'GraphNormal', 'GetWindow', 'GetSelection', 
+			'GetMarquee', 'GetLastUserMenuInfo', 'GetFileFolderInfo', 'GetAxis', 
+			'FuncFitMD', 'FuncFit', 'FTPUpload', 'FTPDownload', 
+			'FTPDelete', 'FTPCreateDirectory', 'FStatus', 'FSetPos', 
+			'FReadLine', 'fprintf', 'FPClustering', 'FindValue', 
+			'FindSequence', 'FindRoots', 'FindPointsInPoly', 'FindPeak', 
+			'FindLevels', 'FindLevel', 'FilterIIR', 'FilterFIR', 
+			'FIFOStatus', 'FIFO2Wave', 'FFT', 'FBinWrite', 
+			'FBinRead', 'FastOp', 'FastGaussTransform', 'Extract', 
+			'ExperimentModified', 'ExecuteScriptText', 'Execute', 'ErrorBars', 
+			'Edit', 'EdgeStats', 'DWT', 'DuplicateDataFolder', 
+			'Duplicate', 'DSPPeriodogram', 'DSPDetrend', 'DrawText', 
+			'DrawRRect', 'DrawRect', 'DrawPoly', 'DrawPICT', 
+			'DrawOval', 'DrawLine', 'DrawBezier', 'DrawArc', 
+			'DrawAction', 'DoXOPIdle', 'DoWindow', 'DoUpdate', 
+			'DoIgorMenu', 'DoAlert', 'DisplayProcedure', 'DisplayHelpTopic', 
+			'Display', 'dir', 'Differentiate', 'DeletePoints', 
+			'DeleteFolder', 'DeleteFile', 'DelayUpdate', 'DefineGuide', 
+			'DefaultGuiFont', 'DefaultGuiControls', 'DefaultFont', 'DebuggerOptions', 
+			'Debugger', 'CWT', 'CustomControl', 'CurveFit', 
+			'Cursor', 'CtrlNamedBackground', 'CtrlFIFO', 'CtrlBackground', 
+			'Cross', 'CreateAliasShortcut', 'Correlate', 'CopyScales', 
+			'CopyFolder', 'CopyFile', 'Convolve', 'ConvexHull', 
+			'ControlUpdate', 'ControlInfo', 'ControlBar', 'Concatenate', 
+			'ColorTab2Wave', 'ColorScale', 'CloseProc', 'CloseMovie', 
+			'Close', 'ChooseColor', 'CheckDisplayed', 'CheckBox', 
+			'Chart', 'cd', 'Button', 'BuildMenu', 
+			'BrowseURL', 'BoundingBall', 'Beep', 'BackgroundInfo', 
+			'AutoPositionWindow', 'AppendXYZContour', 'AppendToTable', 'AppendToLayout', 
+			'AppendToGraph', 'AppendText', 'AppendMatrixContour', 'AppendLayoutObject', 
+			'AppendImage', 'Append', 'APMath', 'AdoptFiles', 
+			'AddMovieFrame', 'AddMovieAudio', 'AddFIFOVectData', 'AddFIFOData', 
+			'Abort'
+			),
+		5 => array(
+			'#pragma', '#include'
+			),
+		),
+	'SYMBOLS' => array(
+		'(', ')', '[', ']', '{', '}', '!', '@', '%', '&', '*', '|', '/', '<', '>'
+		),
+	'CASE_SENSITIVE' => array(
+		GESHI_COMMENTS => false,
+			1 => false,
+			2 => false,
+			3 => false,
+			4 => false,
+			5 => false
+			),
+	'STYLES' => array(
+		'KEYWORDS' => array(
+			1 => 'color: #0000ff;',
+			2 => 'color: #0000ff;',
+			3 => 'color: #c34e00;',
+			4 => 'color: #007575;',
+			5 => 'color: #cc00a3;'
+			),
+		'COMMENTS' => array(
+			1 => 'color: #ff0000; font-style: italic;',
+			),
+		'ESCAPE_CHAR' => array(
+			0 => 'color: #009c00;'
+			),
+		'BRACKETS' => array(
+			0 => 'color: #000000;'
+			),
+		'STRINGS' => array(
+			0 => 'color: #009c00;'
+			),
+		'NUMBERS' => array(
+			0 => 'color: #000000;'
+			),
+		'METHODS' => array(
+			0 => 'color: #000000;'
+			),
+		'SYMBOLS' => array(
+			0 => 'color: #000000;'
+			),
+		'REGEXPS' => array(
+			0 => 'color: #000000;'
+			),
+		'SCRIPT' => array(
+			0 => '',
+			1 => '',
+			2 => '',
+			3 => ''
+			)
+			),
+	'URLS' => array(
+			1 => '',
+			2 => '',
+			3 => '',
+			4 => '',
+			5 => ''
+			),
+	'OOLANG' => false,
+	'OBJECT_SPLITTERS' => array(
+		1 => '',
+		2 => ''
+		),
+	'REGEXPS' => array(
+		0 => "[\$]{1,2}[a-zA-Z_][a-zA-Z0-9_]*",
+		1 => array(
+			GESHI_SEARCH  => "([a-zA-Z]+)(\n)(.*)(\n)(\1;?)",
+			GESHI_REPLACE => '\3',
+			GESHI_BEFORE => '\1\2',
+			GESHI_AFTER => '\4\5',
+			GESHI_MODIFIERS => 'siU'
+			)
+		),
+	'STRICT_MODE_APPLIES' => GESHI_NEVER,
+	'SCRIPT_DELIMITERS' => array(
+		0 => array(
+			'<?php' => '?>'
+			),
+		1 => array(
+			'<?' => '?>'
+			),
+		2 => array(
+			'<%' => '%>'
+			),
+		3 => array(
+			'<script language="igor">' => '</script>'
+			)
+			),
+	'HIGHLIGHT_STRICT_BLOCK' => array(
+		0 => true,
+		1 => true,
+		2 => true,
+		3 => true,
+		4 => true,
+		5 => true
+		)
+);
\ No newline at end of file
