diff --git a/src/VotingApiLoader.php b/src/VotingApiLoader.php
index 60bbc2b..53f1fcb 100644
--- a/src/VotingApiLoader.php
+++ b/src/VotingApiLoader.php
@@ -15,9 +15,11 @@ class VotingApiLoader {
     $definitions = $manager->getDefinitions();
     $entity = \Drupal::service('entity_type.manager')->getStorage($entity_type)->load($entity_id);
     $plugin = $manager->createInstance($plugin_id, $definitions[$plugin_id]);
-    $fieldDefinition = $entity->{$field_name}->getFieldDefinition();
-    if ($fieldDefinition->get('status') != 1) {
-      $read_only = TRUE;
+    if (isset($entity->{$field_name})) {
+      $fieldDefinition = $entity->{$field_name}->getFieldDefinition();
+      if ($fieldDefinition->get('status') != 1) {
+        $read_only = TRUE;
+      }
     }
     return $plugin->buildForm($entity_type, $entity_bundle, $entity_id, $vote_type, $field_name, $style, $show_results, $read_only);
   }
